expr/expr.go

```diff
@@ -3266,6 +3267,82 @@ func EvalExpr(e *expr, from, until int32, values map[MetricRequest][]*MetricData
 		}
 
 		return results, nil
+
+	case "cactiStyle": // cactiStyle(seriesList, system=None, units=None)
+		// Get the series data
+		original, err := getSeriesArg(e.args[0], from, until, values)
+		if err != nil {
+			return nil, err
+		}
+
+		// Get the arguments
+		system, err := getStringNamedOrPosArgDefault(e, "system", 1, "")
+		if err != nil {
+			return nil, err
+		}
+		unit, err := getStringNamedOrPosArgDefault(e, "units", 2, "")
+		if err != nil {
+			return nil, err
+		}
+
+		// Deal with each of the series
+		for _, a := range original {
+			// Calculate min, max, current
+			//
+			// This saves calling summarizeValues 3 times and looping over
+			// the metrics 3 times
+			//
+			// For min:
+			// Ignoring any absent values and inf (if we have a value)
+			// Using summarizeValues("min", ...) results in incorrect values, when absent
+			// values are present
+			//
+			val_min := math.Inf(-1)
+			val_current := math.Inf(-1)
+			val_max := math.Inf(-1)
+			for i, av := range a.GetValues() {
+				if !a.IsAbsent[i] && (val_min == math.Inf(-1) || av < val_min) {
+					val_min = av
+				}
+				if !a.IsAbsent[i] && av > val_max {
+					val_max = av
+				}
+				val_current = av
```

Comment:
I noticed afterwards that val_current isn't guarded against a.IsAbsent[i] - which caused your test to fail:

```
expr_test.go:2168: bad name for cactiStyle() metric 0: got metric1 Current: 0 Max: 5 Min: 1, want metric1 Current: 5 Max: 5 Min: 1
```